### PR TITLE
Don't embed bitcode on tvOS anymore

### PR DESCRIPTION
**Pull request by filipnavara**
Apple effectively deprecated bitcode embedding with Xcode 14 and no longer requires it for App Store submissions.

https://developer.apple.com/documentation/Xcode-Release-Notes/xcode-14-release-notes:
> Starting with Xcode 14, bitcode is no longer required for watchOS and tvOS applications, and the App Store no longer accepts bitcode submissions from Xcode 14.
>
> Xcode no longer builds bitcode by default and generates a warning message if a project explicitly enables bitcode: “Building with bitcode is deprecated. Please update your project and/or target settings to disable bitcode.” The capability to build with bitcode will be removed in a future Xcode release. IPAs that contain bitcode will have the bitcode stripped before being submitted to the App Store. Debug symbols can only be downloaded from App Store Connect / TestFlight for existing bitcode submissions and are no longer available for submissions made with Xcode 14. (86118779)

### CODE REVIEW

**Review by rolfbjarne — APPROVED** [review]
This should shrink the runtime pack significantly!

**Comment by MichalStrehovsky** [comment]
We can also undo the ifdef from https://github.com/dotnet/runtime/pull/79322.

**Comment by filipnavara** [comment]
> We can also undo the ifdef from #79322.

Done

**Review by steveisok — APPROVED** [review]
Thanks!

File: eng/native/configurecompiler.cmake [hunk 1]
```diff
@@ -482,9 +482,7 @@ if (CLR_CMAKE_HOST_UNIX)
   
   # Separate functions so linker can remove them. But not on tvOS because
   # -ffunction-sections is not supported with -fembed-bitcode.
```
Comment (MichalStrehovsky) [inline eng/native/configurecompiler.cmake:484]:
```suggestion
  # Separate functions so linker can remove them.
```

**Pull request merged**